cmd/sourcepods-dev/dev_setup.go

```diff
@@ -57,6 +63,33 @@ func devSetupAction(c *cli.Context) error {
 
 	return nil
 }
+func genKeys() error {
```

Comment:
We could probably do this in Go, but I don't know how :joy:

Reply:
I could give it a try after merging the pr... Do we want to store them on the fs? Or possibly in the db? Than we could launch the ssh server stateless.

Reply:
Or maybe on the storage service

Reply:
This is purely for dev-setup, for production setups you'd inject them into your container "somehow". Don't see a point in having them in the DB, they _should_ not change that frequently. (They're HostKeys...)

Reply:
Ideally they should never change after the setup, otherwise ssh will complain. But they got to be stored somewhere, and if they are put on the fs you always need a volume just for that.

Reply:
no volume required, this is _only_ run for `sourcepods-dev`, so for production this code is _never_ executed :)

Reply:
Actually, seems like doing it in Go would be _more_ code, not less :P https://github.com/go-gitea/gitea/blob/7ae59567a2ec3065b10b75aff545edb06aac4790/modules/ssh/ssh.go#L204-L238

Reply:
Whenever possible we should try to write things in Go, even if it's a bit more code, I don't mind it.
If possible, I would like to have the opportunity to have all keys within a processes memory and do reload in memory, when those keys change. We should be able to trigger those reloads via APIs in the future.
That way scaling the ssh daemon / server should also be easier.

Reply:
I wanna introduce pub-sub for this :trollface: have `PubkeyAdded(pubkey)`/`PubkeyRemoved(pubkey)`-events, and only reload on `PubKeyReload()` or service startup :thinking: